2021/day11/day11.java

```diff
@@ -0,0 +1,164 @@
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class day11 {
+    public static class Point {
+        int x;
+        int y;
+
+        Point(int horizontal, int vertical) {
+            x = horizontal;
+            y = vertical;
+        }
+    }
+
+    public static ArrayList<ArrayList<Integer>> readInput(String filename) {
+        ArrayList<ArrayList<Integer>> octoGrid = new ArrayList<>();
+        Scanner sc;
+        try {
+            sc = new Scanner(new File(filename));
+            sc.useDelimiter("");
+            while (sc.hasNextInt()) {
+                ArrayList<Integer> octoRow = new ArrayList<>();
+                while (sc.hasNextInt()) {
+                    octoRow.add(sc.nextInt());
+                }
+                octoGrid.add(octoRow);
+                if (sc.hasNextLine()) {
+                    sc.nextLine();
+                }
+            }
+            sc.close();
+        } catch (Exception e) {
+            System.out.println("Something went wrong " + e);
+        }
+        return octoGrid;
+    }
+
+    public static ArrayList<Point> getNeighbours(int x, int y, int width, int height) {
+        ArrayList<Point> neighbours = new ArrayList<>();
+        if (x > 0) {
+            neighbours.add(new Point(x - 1, y));
+            if (y > 0) {
+                neighbours.add(new Point(x - 1, y - 1));
+            }
+            if (y < height - 1) {
+                neighbours.add(new Point(x - 1, y + 1));
+            }
+        }
+        if (x < width - 1) {
+            neighbours.add(new Point(x + 1, y));
+            if (y > 0) {
+                neighbours.add(new Point(x + 1, y - 1));
+            }
+            if (y < height - 1) {
+                neighbours.add(new Point(x + 1, y + 1));
+            }
+        }
+        if (y > 0) {
+            neighbours.add(new Point(x, y - 1));
+        }
+        if (y < height - 1) {
+            neighbours.add(new Point(x, y + 1));
+        }
+        return neighbours;
+    }
```

Comment:
This is too long and too repetitive - if it was only 4 neighbours as I said in an earlier comment, I would accept such a solution, but with 8 it's too long. You can either, as I said before, write an `isInBounds` method or you can even implicitly fit it in here (don't include the comments ofc):

```java
for (int nx = Math.max(0, x - 1); nx < Math.min(width, x + 2); ++nx) {  // Implicitly checking the bounds for x
  for (int ny = Math.max(0, y - 1); ny < Math.min(height, y + 2); ++ny) {  // Implicitly checking the bounds for y
    if (nx != x || ny != y) {  // Omit the point itself
      neighbours.add(new Point(nx, ny));
    }
  }
}
```